### PR TITLE
[QoS][JSONRPC] Implement proper null ID handling for JSON-RPC 2.0 specification compliance

### PR DESCRIPTION
**Pull request by adshmh**
## Summary

Refactor JSON-RPC ID type to properly handle null values per JSON-RPC 2.0 specification

### Primary Changes:

- Convert ID struct fields from values to pointers (*int, *string) to distinguish between zero/empty values and null
- Update ID.MarshalJSON() and ID.UnmarshalJSON() to properly handle null JSON values according to specification
- Modify Request.MarshalJSON() to always include id field (serializes as null when unset) for JSON-RPC 2.0 compliance
- Add IDFromInt() and IDFromStr() constructor functions for creating properly initialized ID instances

### Secondary Changes:

- Add comprehensive test cases for edge cases like null IDs, zero values, and empty strings
- Update existing test expectations to reflect new null ID behavior
- Improve code documentation with specification references and clearer comments
- Fix typo in test case name ("speicifed" → "specified")

## Issue

Some users send JSONRPC requests with no ID field, while some servers return an empty response to this type of request

- Issue or PR: #{ISSUE_OR_PR_NUMBER}

## Type of change

Select one or more from the following:

- [x] New feature, functionality or library
- [ ] Bug fix
- [ ] Code health or cleanup
- [ ] Documentation
- [ ] Other (specify)

### E2E Validation & Tests

- [ ] `make path_up`
- [ ] `make test_e2e_evm_shannon`
- [ ] `make test_e2e_evm_morse`

### Observability

- [ ] 1. `make path_up`
- [ ] 2. Run one of the following:
  - For `Shannon` with `anvil`: `make test_request__shannon_relay_util_100`
  - For `Morse` with `F00C`: `make test_request__morse_relay_util_100`
- [ ] 3. Visit [PATH Relay Grafana Dashboard](http://localhost:3003/d/relays/path-service-requests) to view results

## Sanity Checklist

- [ ] I have updated the GitHub Issue `assignees`, `reviewers`, `labels`, `project`, `iteration` and `milestone`
- [ ] For docs, I have run `make docusaurus_start`
- [ ] For code, I have run `make test_all`
- [ ] For configurations, I have updated the documentation
- [ ] I added `TODO`s where applicable

### CODE REVIEW

File: qos/jsonrpc/request.go [hunk 1]
```diff
@@ -13,43 +13,44 @@ type Version string
...
+// Specification requirements:
+//   - jsonrpc: must be "2.0"
+//   - method: string containing the method name
+//   - params: structured values (array or object), optional
+//   - id: identifier for correlation, always included (null if unset)
+//
+// Reference: https://www.jsonrpc.org/specification#request_object
 type Request struct {
-	ID      ID      `json:"id,omitempty"`
+	ID      ID      `json:"id"` // Always include in JSON
```
Comment (Olshansk) [inline qos/jsonrpc/request.go:26]: 1. #PUC why
2. #PUC how we handle empty IDs

File: qos/jsonrpc/id.go [hunk 2]
```diff
@@ -5,64 +5,88 @@ import (
...
+//   - Must be String, Number, or NULL
+//   - Server must echo the same value in Response
+//   - Should not be NULL in normal operation
+//   - Numbers must not contain fractional parts
+//
+// Reference: https://www.jsonrpc.org/specification#id
 type ID struct {
-	intID int
-	strID string
+	intID *int
```
Comment (Olshansk) [inline qos/jsonrpc/id.go:18]: #PUC why this is a pointer and not a value

**Review by Olshansk — CHANGES_REQUESTED** [review]
